mobius/src/transformable/collection.rs

```diff
@@ -20,12 +21,25 @@ impl<T> Collection<T> {
     pub fn get_primitives(&self) -> &[T] {
         &self.primitives
     }
+}
 
-    pub fn union(primitives: Vec<Self>) -> Self {
-        let all_primitives: Vec<T> = primitives.into_iter().flat_map(|x| x.primitives).collect();
-        Self {
-            primitives: all_primitives,
-        }
+impl<T: Clone> Mul for Collection<T> {
```

Comment:
Add some documentation to explain that the point of this is to implement Semigroup so you can combine collections into one big collection